### DISCUSSION TITLE
How to increase wsl2 disk space ?

### DISCUSSION THREAD

**Discussion by DachuanZhao** [microsoft/WSL#6679] [Q&A]
**Is your feature request related to a problem? Please describe.**
I want to increase wsl2 disk space because I run a mysql in wsl2 and its performance is affected by disk allocation .

**Describe the solution you'd like**
I run 
```Powershell
wsl.exe --shutdown
resize-vhd -Path .\ext4.vhdx -SizeBytes 300GB
``` 
But it doesn't seem to work ...

**Comment by WSLUser** [comment]
Use C:\Users\username\ .wslconfig. Check the MS WSL Docs.

**Comment by craigloewen-msft** [comment]
@DachuanZhao could you try following the instructions here and letting us know if that solves your issue:
https://docs.microsoft.com/en-us/windows/wsl/compare-versions#expanding-the-size-of-your-wsl-2-virtual-hard-disk

Thanks!

**Reply by MurrMack** [reply]
New link for anyone stumbling on this - https://docs.microsoft.com/en-us/windows/wsl/vhd-size

**Comment by jgresham** [comment]
Is there a way to set the default disk space in the `wsl.conf` or `.wslconfig` file? It would be nice to have this to avoid having to run all the commands. Docs https://learn.microsoft.com/en-us/windows/wsl/wsl-config
